File: sprint6/By languages/C#/lesson11code3.cs
// Вызываем функцию поиска пути
var (dist, currentPoint) = escapeFromMaze(n, m, field, start_point);
List<(int, int)> path = new List<(int, int)>();

// Строим путь от текущей точки до стартовой точки
while (currentPoint != start_point) {
    path.Add(currentPoint);
    currentPoint = prev[currentPoint];
}

// Условие выхода из цикла составлено так, что начальная точка не будет добавлена в путь, поправим это.
path.Add(start_point);

// Разворачиваем путь, так как первой мы добавляли конечную точку маршрута.
path.Reverse();

File: sprint6/By languages/swift/lesson11code3.swift
// Вызываем функцию поиска пути
let (dist, currentPoint) = escapeFromMaze(n: n, m: m, field: field, startPoint: start_point)
var path = [Point]()

// Строим путь от текущей точки до стартовой точки
var current = currentPoint
while current != start_point {
    path.append(current)
    current = Point(x: prev[current.x][current.y] / m, y: prev[current.x][current.y] % m)
}

// Добавляем начальную точку в путь
path.append(start_point)

// Разворачиваем путь, так как первой мы добавляли конечную точку маршрута.
path.reverse()
